tests/reboot/reboot_test.go

```diff
@@ -104,6 +105,71 @@ var _ = Describe("{RebootOneNode}", func() {
 	})
 })
 
+var _ = Describe("{ReallocateSharedMount}", func() {
+	var contexts []*scheduler.Context
+
+	It("has to schedule apps and reboot node(s) with shared volume mounts", func() {
+		//var err error
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().ScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("reallocate-mount-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+
+		Step("get nodes with shared mount and reboot them", func() {
+			for _, ctx := range contexts {
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for _, vol := range vols {
+					if vol.Shared {
+						n, err := Inst().V.GetNodeForVolume(vol, 1*time.Minute, 5*time.Second)
+						Expect(err).NotTo(HaveOccurred())
+						logrus.Infof("volume %s is attached on node %s [%s]", vol.ID, n.SchedulerNodeName, n.Addresses[0])
+						StopVolDriverAndWait([]node.Node{*n})
+						err = Inst().V.StopDriver([]node.Node{*n}, false, nil)
+						Expect(err).NotTo(HaveOccurred())
+						err = Inst().N.RebootNode(*n, node.RebootNodeOpts{
+							Force: true,
+							ConnectionOpts: node.ConnectionOpts{
+								Timeout:         1 * time.Minute,
+								TimeBeforeRetry: 5 * time.Second,
+							},
+						})
+						Expect(err).NotTo(HaveOccurred())
+						logrus.Infof("waiting for 20 seconds to node %s to reboot", n.SchedulerNodeName)
+						time.Sleep(20 * time.Second)
+						// as we keep the storage driver down on node until we check if the volume we need to force
+						// driver to refresh endpoint to pick aanother storage node which is up
+						ctx.RefreshStorageEndpoint = true
+						ValidateContext(ctx)
+						n2, err := Inst().V.GetNodeForVolume(vol, 1*time.Minute, 5*time.Second)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(n2.SchedulerNodeName).NotTo(Equal(n.SchedulerNodeName))
+						logrus.Infof("volume %s is now attached on node %s [%s]", vol.ID, n2.SchedulerNodeName, n2.Addresses[0])
+						StartVolDriverAndWait([]node.Node{*n})
+						err = Inst().S.EnableSchedulingOnNode(*n)
```

Comment:
I'm not quite understand why do we need  EnableSchedulingOnNode here, you didn't disable it before.

Reply:
good catch it seems I removed it mistakenly during last rebase